ontopy/ontology.py

```diff
@@ -1490,22 +1499,23 @@ def addancestors(entity, counter, subject):
                     subject.add(parent)
                     addancestors(parent, counter - 1, subject)
 
-        if isinstance(include, str) and include.isdigit():
-            include = int(include)
-
-        if include == "all":
-            ancestors.update(*(_.ancestors() for _ in classes))
-        elif include == "closest":
-            closest = self.closest_common_ancestor(*classes)
+        if sum(map(bool, [closest, generations])) > 1:
```

Comment:
Easier to read if you write
```python
        if closest and generations is not None:
```
Another benefit is that it handles the case of setting `generations=0` correctly.

It will be slightly more efficient (with no cost in readability) if you place this check within the if-statement on line 1506, like:
```python
if closest:
    if generations is not None:
        raise ValueError(...)
    closest_ancestor = ...
    ...
```